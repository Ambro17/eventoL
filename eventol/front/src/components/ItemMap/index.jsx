import React from 'react';
import PropTypes from 'prop-types'
import './index.scss';


export default class ItemMap extends React.Component {
  propTypes = {
    place: PropTypes.string,
    title: PropTypes.string,
    url: PropTypes.string,
    attendees: PropTypes.string,
    overview: PropTypes.string,
    uid: PropTypes.string,
    sliderId: PropTypes.string
  };

  getMapId(){
    const {uid, sliderId} = this.props;
    return `${sliderId}${uid}`;
  }

  loadMap(){
    const {place} = this.props;
    const {geometry} = JSON.parse(place);
    const event_lat = geometry.location.lat;
    const event_lon = geometry.location.lng;
    const mapId = this.getMapId();
    const map = L.map(mapId, {
        scrollWheelZoom: false,
        zoomControl    : false,
        dragging       : false,
        keyboard       : false,
        doubleClickZoom: false,
        touchZoom      : false,
        boxZoom        : false,
        minZoom: 11,
        maxZoom: 11
    }).setView([event_lat, event_lon], 5);
    L.tileLayer('https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png', {
        attribution: ''
    }).addTo(map);
    map.attributionControl.setPrefix('');
    L.marker([event_lat, event_lon]).addTo(map);
  }

  componentDidMount(){
    const uid = this.getMapId();
    try {
      this.loadMap();
    } catch(e){}
    document.getElementById(uid).classList.remove('max-size')
  }

  render(){
    const {title, url, attendees, overview, uid, sliderId} = this.props;
    const mapId = this.getMapId();
    return (
      <div id={mapId} className="Item max-size">
        <a href={url}>
          <div className="overlay">
            <div className="title">{title}</div>
            <div className="rating">{gettext('Attendees')}: {attendees}</div>
            <div className="plot">{overview}</div>
          </div>
        </a>
      </div>
    )
  }
}
