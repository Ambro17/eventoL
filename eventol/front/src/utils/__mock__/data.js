export const event1 = {
  attendees: 10,
  backdrop: 'event1backdrop',
  eventSlug: 'event1slug',
  overview: 'event1overview',
  place: 'event1place',
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 1',
  url: 'event1Url',
  id: 1,
};

export const event2 = {
  attendees: 25,
  backdrop: 'event2backdrop',
  eventSlug: 'event2slug',
  overview: 'event2overview',
  place: 'event2place',
  tags: [
    {
      name: 'slug-name',
      slug: 'slug-slug',
    },
  ],
  title: 'event 2',
  url: 'event2Url',
  id: 2,
};

export default {
  event1,
  event2,
};
