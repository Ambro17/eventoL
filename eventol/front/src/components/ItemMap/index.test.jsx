import React from 'react';
import renderer from 'react-test-renderer';

import ItemMap from '.';

import {event1} from '../../utils/__mock__/data';

describe('ItemMap', () => {
  let component;
  let sliderId;
  let tree;

  const getComponent = () => {
    component = renderer.create(<ItemMap sliderId={sliderId} {...event1} />);
    return component;
  };

  beforeEach(() => {
    sliderId = event1.id.toString();
    const id = `${sliderId}${event1.eventSlug}`;
    document.body.innerHTML = `<div><div id="${id}">Map</div></div>`;
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
