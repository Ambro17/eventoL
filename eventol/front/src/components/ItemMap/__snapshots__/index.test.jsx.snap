// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemMap Default Snapshot 1`] = `undefined`;
